### ISSUE TITLE
"DELETE" button on numeric keyboard on iOS does not have translation on VoiceOver

### ISSUE THREAD

**Issue by s4nchez** [flutter/flutter#173340]
### Steps to reproduce

1. Configure iOS to use a different language than English (tried with Italian and Portuguese)
2. Turn on VoiceOver in Accessibility settings
3. Open a numeric text input (`keyboardType: TextInputType.number`) on any flutter app
4. Hit the delete (⌫) key

### Expected results

The voice over should say the "delete" word in the local language (e.g. "Apagar" in Portuguese, or "Eliminare" in Italian).

### Actual results

The voice over says "delete" (English word).

Please notice that:
1. The behaviour is correct in non-flutter apps (e.g. wi-fi -> proxy settings -> manual -> port).
2. All numbers are translated correctly.

### Code sample

<details open><summary>Code sample</summary>

```dart
TextField(
  keyboardType: TextInputType.number,
),
```
</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.32.8, on macOS 15.5 24F74 darwin-arm64, locale en-GB) [1,497ms]
    • Flutter version 3.32.8 on channel stable at /Users/isz898/fvm/versions/3.32.7
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision edada7c56e (12 days ago), 2025-07-25 14:08:03 +0000
    • Engine revision ef0cd00091
    • Dart version 3.8.1
    • DevTools version 2.45.1

[!] Android toolchain - develop for Android devices (Android SDK version 35.0.1) [2.8s]
    • Android SDK at /Users/isz898/Library/Android/sdk
    • Platform android-35, build-tools 35.0.1
    • ANDROID_HOME = /Users/isz898/Library/Android/sdk
    • Java binary at: /Users/isz898/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.5+-13047016-b750.29)
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 16.4) [2.0s]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16F6
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [115ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.3) [115ms]
    • Android Studio at /Users/isz898/Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.5+-13047016-b750.29)

[✓] IntelliJ IDEA Ultimate Edition (version 2025.1.1.1) [114ms]
    • IntelliJ at /Users/isz898/Applications/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 86.0.2
    • Dart plugin version 251.27623.5

[✓] IntelliJ IDEA Community Edition (version 2024.3.5) [37ms]
    • IntelliJ at /Users/isz898/Applications/IntelliJ IDEA Community Edition.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.102.3) [9ms]
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (5 available) [6.6s]
    • sdk gphone64 arm64 (mobile) • emulator-5554                        • android-arm64  • Android 16 (API 36) (emulator)
    • Ivan's iPhone (mobile)      • 00008110-000E45380A98401E            • ios            • iOS 18.5 22F76
    • iPhone 16 Pro (mobile)      • AEB58BDE-2EA1-4168-A9E9-EE3D9388B1F5 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-18-3
      (simulator)
    • macOS (desktop)             • macos                                • darwin-arm64   • macOS 15.5 24F74 darwin-arm64
    • Chrome (web)                • chrome                               • web-javascript • Google Chrome 138.0.7204.184
    ! Error: Browsing on the local area network for Ivan’s iPad. Ensure the device is unlocked and attached with a cable or associated with
      the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources [653ms]
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by tirth-patel-nc** [comment]
thanks for the report. we're seeing the same behavior with the latest SDK versions on both Android and iOS. my colleague verified it on iOS, and I confirmed it on Android by changing the system language to Gujarati, the **DELETE** action was still announced in English.

```
stable : 3.32.8
master : 3.33.0-1.0.pre-1299
```

#154537, #99600 might be related or possibly dupe.

**Comment by chunhtai** [comment]
IIRC flutter doesn't own this keyboard and it is draw by the system ui. either way this seems like belongs to text input team

**Comment by s4nchez** [comment]
@chunhtai that's not entirely accurate. While the keyboard itself is indeed draw and managed by iOS once it's open, Flutter configures it, and I suspect there's where the problem lies. Otherwise, it wouldn't behave differently from other non-Flutter apps, which is the case here.

**Comment by tirth-patel-nc** [comment]
It is replicable on Android too. https://github.com/flutter/flutter/issues/173340#issuecomment-3160000927

**Comment by loic-sharma** [comment]
cc @LongCatIsLooong to see if this reproduces in a UIKit app

**Comment by LongCatIsLooong** [comment]
Not reproducible on iOS18.6. 

> The voice over should say the "delete" word in the local language (e.g. "Apagar" in Portuguese, or "Eliminare" in Italian).

From what I'm seeing on iOS18.6, this isn't true. Each keyboard typically uses the keyboard language instead of system locale to determine the accessibility announcement language. You'll have to switch to the right keyboard.

I'm guessing the Portuguese / Italian keyboards do not have a numpad version so the english numpad is used, thus the English announcement.

**Comment by LongCatIsLooong** [comment]
> wi-fi -> proxy settings -> manual -> port

You can change the numpad's keyboard language to Italian in a system settings text field but that doesn't seem to be available for regular UIKit apps. When I press the globe icon at the bottom left of the keyboard I only get Arabic and French for numpads.
